R/data.table.R

```diff
@@ -1887,14 +1887,19 @@ chmatch2 <- function(x, table, nomatch=NA_integer_) {
 #    x
 #}
 
-as.matrix.data.table <- function(x, rownames, ...) {
+as.matrix.data.table <- function(x, rownames, rownames.literal=FALSE, ...) {
```

Comment:
In such unwanted edge case we don't have to be consistent to data.frame by default, it is important to give option for consistency to data.frame, which new arg already provides, but we could tune its default.
Maybe new arg could have smart default like `nrow(x)==1L`? probably more correct would be `nrow(x)==1L && !missing(rownames) && is.character(rownames)` but it is a little bit messy, maybe `nrow(x)==1L` is sufficient in this case.